src/ripple/app/tx/impl/OfferStream.cpp

```diff
@@ -222,6 +291,66 @@ TOfferStreamBase<TIn, TOut>::step()
                     << "Removing became unfunded offer " << entry->key();
             }
             offer_ = TOffer<TIn, TOut>{};
+            // See comment at top of loop for why the offer is removed
+            continue;
+        }
+
+        bool const rmSmallIncreasedQOffer = [&] {
+            bool const inIsXRP = isXRP(offer_.issueIn());
+            bool const outIsXRP = isXRP(offer_.issueOut());
+            if (inIsXRP && !outIsXRP)
+            {
+                // Without the `if constexpr`, the
+                // `shouldRmSmallIncreasedQOffer` template will be instantiated
+                // even if it is never used. This can cause compiler errors in
+                // some cases, hense the `if constexpr` guard.
```

Comment:
Great comment!  Thanks!  BTW, perhaps change "hense" to "hence'?

Reply:
fixed in `9b8c050cb3 [fold] Minor cleanups`